homeassistant/components/zeroconf/__init__.py

```diff
@@ -395,6 +403,7 @@ def async_service_update(
         )
 
         if state_change == ServiceStateChange.Removed:
```

Comment:
This one has been a todo for years 😄

Reply:
Great timing, we just fixed ESPHome to send proper un-announce https://github.com/esphome/esphome/pull/4399